include/ginkgo/core/stop/residual_norm.hpp

```diff
@@ -49,47 +49,236 @@ namespace stop {
 
 
 /**
- * The ResidualNorm class provides a framework for stopping criteria
+ * The mode for the residual norm criterion.
+ *
+ * - absolute:        Check for tolerance against residual norm.
+ *                    || r || < tau
+ *
+ * - initial_resnorm: Check for tolerance relative to the initial residual norm.
+ *                    || r ||/|| r_0|| < tau
+ *
+ * - rhs_resnorm:     Check for tolerance relative to the rhs norm.
+ *                    || r ||/|| b || < tau
```

Comment:
Can you check how this is formatted in Doxygen? It might be helpful to add $ $ around the mathematical expressions

Reply:
I added the $ $ and frac to make it prettier in doxygen.